src/srcs.c

```diff
@@ -114,7 +115,12 @@ static double get_rvel(ParamCoLoRe *par,int ix,int iy,int iz,
   else
     v[2]=par->grid_npot[ix_0+iy_0+iz_hi]-par->grid_npot[ix_0+iy_0+iz_lo];
 
-  return 0.5*idx*(v[0]*u[0]+v[1]*u[1]+v[2]*u[2]);
+  *rvel = 0.5*idx*(v[0]*u[0]+v[1]*u[1]+v[2]*u[2]);
+#ifdef TVEL
+  *vtheta = 0.5*idx*(v[0]*(u[0]*u[2]/sqrt(1-u[2]*u[2])) + v[1]*(u[1]*u[2]/sqrt(1-u[2]*u[2])) + v[2]*(-sqrt(1-u[2]*u[2])));
```

Comment:
It might be worth using [cart2sph](https://github.com/damonge/CoLoRe/blob/d19ef73ef049d772918edb38c69ce632542cff6c/src/srcs.c#L68) to avoid potential divisions by zero here (and potentially some performance boost)